app/models/manageiq/providers/azure/cloud_manager/vm/operations/power.rb

```diff
@@ -3,8 +3,8 @@ module ManageIQ::Providers::Azure::CloudManager::Vm::Operations::Power
 
   included do
     supports :reboot_guest do
-      unsupported_reason_add(:reboot_guest, unsupported_reason(:control)) unless supports?(:control)
-      unsupported_reason_add(:reboot_guest, _("The VM is not powered on")) unless current_state == "on"
+      unsupported_reason(:control) ||
+        (_("The VM is not powered on") unless current_state == "on")
```

Comment:
The `|| (... unless condition)` works but looks overly complicated (I missed the added `()` at first)
What about using a ternary operator here? e.g.: `current_state == "on" ? unsupported_reason(:control) : _("The VM is not powered on")`
I assume `unsupported_reason(:control)` will return `nil` if it is supported right?

Reply:
of note, this is technically different.
if current_state is off, and unsupported reason is false, then the error message coming out will be different.

But that is only a test failure issue, not a real issue.